### PR TITLE
Change Tags to Roles 

### LINKED ISSUE #49: Fix regex validation for `Telegram`
Telegram string should not contain whitespace
To fix `VALIDATION_REGEX` in `Telegram.java` to conform with this specification

### LINKED ISSUE #62: Change tags to roles
- [x] Change occurrences of tags to roles
- [x] Change regex validation for tags to reflect roles requirements
- [x] Change tests to reflect roles requirements

### PR DESCRIPTION
**Pull request by shizy**
Fixes #49, Fixes #62 
- Change all occurrences of tags to roles
- Add restrictions for the add fields, as per feature specifications. This includes telegram and role regex validations
- Update error messages for invalid add/edit commands

### CODE REVIEW

File: src/test/java/seedu/address/model/person/NameContainsKeywordsPredicateTest.java [hunk 1]
```diff
@@ -68,9 +68,9 @@ public void test_nameDoesNotContainKeywords_returnsFalse() {
         predicate = new NameContainsKeywordsPredicate(Arrays.asList("Carol"));
         assertFalse(predicate.test(new PersonBuilder().withName("Alice Bob").build()));
 
-        // Keywords match phone, email and telegram, but does not match name
+        // Keywords match phone, email and name, but does not match name
```
Comment (yyihaoc) [inline src/test/java/seedu/address/model/person/NameContainsKeywordsPredicateTest.java]: is this supposed to be telegram instead of name?

**Review by yyihaoc — COMMENTED** [review]
LGTM

File: src/test/java/seedu/address/model/person/NameContainsKeywordsPredicateTest.java (hunk 1 above)
Comment (shizy) [reply src/test/java/seedu/address/model/person/NameContainsKeywordsPredicateTest.java]: Minor oversight, fixed in subsequent commit

**Review by superb-sushi — APPROVED** [review]
LGTM

**Review by yyihaoc — APPROVED** [review]
LGTM!

**Pull request merged**